### PR TITLE
Added sentinel node implementation

### PR DESCRIPTION
**Pull request by ChinthanBD**
Added Sentinel node implementation

### CODE REVIEW

File: doubly_linked_list.py [hunk 1]
```diff
@@ -3,58 +3,107 @@
...
-            tail.next_node = Node(key=None, value=value, next_node=None, prev_node=tail)
-        return self.head
+            new_node = Node(key=None, value=value, next_node=None, prev_node=tail)
 
-    def delete_node(self, value):
+            new_node.prev_node = tail.prev_node
+            tail.prev_node.next_node = new_node
+            new_node.next_node = tail
+            tail.prev_node = new_node
+            self.node_count += 1
```
Comment (tejas-k3) [inline doubly_linked_list.py]: Repetitive code is present, it can be refactored

File: doubly_linked_list.py [hunk 2]
```diff
@@ -3,58 +3,107 @@
...
+        head = self.fetch_head()
+        tail = self.fetch_tail()
 
-    def add_node(self, value):
-        if self.head is None:
-            self.head = Node(key=None, value=value, next_node=None, prev_node=None)
+        if self.node_count == 0:
+            new_node = Node(key=None, value=value, next_node=None, prev_node=None)
+            new_node.next_node = head.next_node
+            new_node.prev_node = head.next_node.prev_node
```
Comment (tejas-k3) [inline doubly_linked_list.py]: With sentinel node approach... the  part will be
```
head.next = tail
tail.prev = head
new_node.prev_node = head.next_node.prev_node == tail.prev === head
new_node.prev_node = head

```

File: doubly_linked_list.py [hunk 3]
```diff
@@ -3,58 +3,107 @@
...
+            tail.prev_node = new_node
+            self.node_count += 1
+            return True
+
+    def delete(self, value):
         temp = self.fetch_head()
-        while temp.next_node and temp:
+        #[1]
+        while temp:
             if temp.value == value:
```
Comment (tejas-k3) [inline doubly_linked_list.py:35]: This code will fail for `linked_list.delete(None)` as that's your sentinel node!!!

File: doubly_linked_list.py [hunk 4]
```diff
@@ -3,58 +3,107 @@
...
-                temp.prev_node.next_node = temp.next_node
-                temp.next_node.prev_node = temp.prev_node
-                break
+                if temp.prev_node:
+                    temp.prev_node.next_node = temp.next_node
+                if temp.next_node:
+                    temp.next_node.prev_node = temp.prev_node
```
Comment (tejas-k3) [inline doubly_linked_list.py]: As now you've sentinel nodes.... Your linked_list will Look as below
```
HEAD <--> TAIL  # for empty linked list
HEAD <--> 1 <--> TAIL # for linkedlist with 1 element
HEAD <--> 1 <--> 2 <--> .... <-->TAIL # for linkedlist with many elements
```
`.prev_node` & `next_node` will always be TRUE thanks to sentinel nodes, do you need those checks ?

File: doubly_linked_list.py [hunk 5]
```diff
@@ -3,58 +3,107 @@
...
     print("Doubly Linked List:")
     doubly_linked_list.print_doubly_linked_list()
 
     # Deleting a node from the linked list
     value_to_delete = 20
-    doubly_linked_list.delete_node(value_to_delete)
+    doubly_linked_list.delete(value_to_delete)
 
     # Displaying the linked list after deletion
     print(f"\nDoubly Linked List after deleting node with value {value_to_delete}:")
```
Comment (tejas-k3) [inline doubly_linked_list.py]: Missing tests code for all methods! Add for missing methods POSITIVE and NEGATIVE cases. use CHATGPT

**Review by tejas-k3 — CHANGES_REQUESTED** [review]
Address all review comments

File: doubly_linked_list.py (hunk 3 above)
Comment (ChinthanBD) [reply doubly_linked_list.py:35]: fixed

File: doubly_linked_list.py (hunk 1 above)
Comment (ChinthanBD) [reply doubly_linked_list.py]: Done

File: doubly_linked_list.py (hunk 2 above)
Comment (ChinthanBD) [reply doubly_linked_list.py]: handled

File: doubly_linked_list.py (hunk 4 above)
Comment (ChinthanBD) [reply doubly_linked_list.py]: removed unwanted checks

File: doubly_linked_list.py (hunk 5 above)
Comment (ChinthanBD) [reply doubly_linked_list.py]: Added more cases

Comment (tejas-k3) [reply doubly_linked_list.py]: Where is method for `get_element` and `get_position` ??? Both positive and negative??

**Review by tejas-k3 — CHANGES_REQUESTED** [review]
Missing cases to be added
